src/qibo_cloud_backends/braket_client.py

```diff
@@ -0,0 +1,133 @@
+import networkx as nx
+from braket.devices import LocalSimulator
+from qibo import Circuit as QiboCircuit
+from qibo.backends import NumpyBackend
+from qibo.config import raise_error
+from qibo.result import MeasurementOutcomes
+
+# Import Qiskit packages for transpiler
+from qiskit import QuantumCircuit, qasm2, transpile
+
+from qibo_cloud_backends.braket_translation import to_braket
+
+
+class BraketClientBackend(NumpyBackend):
+    def __init__(self, device=None, verbatim_circuit=False):
```

Comment:
I am assuming that `device` here is an `AwsDevice`, I would rather ask the user to pass the name of the device, for instance `"arn:...qpu/rigetti/Aspen-8"` and then internally build the device as an attribute of the backend. Ideally, the user should not have to deal directly with any of the AWS sdk API, but only with the qibo native API as much as possible.

The thing that is not clear to me, though, is how do you connect to your AWS account. I mean, isn't there any token or key you have to provide when submitting jobs to aws? Anything that identifies who you are?

Reply:
This is how we intend for the device to be used:
``` python
from qibo_cloud_backends.braket_client import BraketClientBackend

# Select device
device = LocalSimulator("default")
# device = AwsDevice("arn:aws:braket:::device/quantum-simulator/amazon/sv1")
# device = AwsDevice('arn:aws:braket:eu-north-1::device/qpu/iqm/Garnet')

# Initialize device
AWS = BraketClientBackend(device = device, verbatim_circuit=True)

# Execute circuit `circuit_qibo`
results = AWS.execute_circuit(circuit_qibo, nshots=1000)
print(results.frequencies())
```
If I don't misunderstand, your suggestion is to pass `device = "arn..."` instead of  `device = AwsDevice("arn...")`, right? If yes, then I suppose that will do too. I can make the change in the next iteration.

As for the AWS account, it's quite complicated (to me, at least). What I did was to 
1. Create an AWS account, log in, and enable certain AWS policies. Then, get the access keys and secret access keys.
2. In Terminal, install the AWS dependencies:
`pip install amazon-braket-sdk boto3 awscli`
3. In Terminal, configure the account:
`aws configure`, input access key, input secret access key, input region, etc.

After that, there's no need to further input tokens or keys when submitting jobs to AWS. 

Pinging @speller26! Cody, could I trouble you to confirm if this is the way to set up the AWS account please? Thank you very much!

Reply:
Yes precisely, I believe it would be better to just ask the user for the name of the device and then take care internally of its construction.

I see, thanks for the explanation. I guess that `aws configure` is probably writing some environment variable that are later silently read when you call `device.run`. Therefore, for now it is fine to have the backend constructed without any identification key (we should point this out explicitely in the docs though, maybe adding a reference to the AWS page showing how to set up your account), but I wonder if it would be possible to switch between different accounts by overwriting the environment variables or directly passing the key/token to the `device`.

Reply:
> Yes precisely, I believe it would be better to just ask the user for the name of the device and then take care internally of its construction.

Sure, I have amended it in the latest push.

> I see, thanks for the explanation. I guess that `aws configure` is probably writing some environment variable that are later silently read when you call `device.run`. Therefore, for now it is fine to have the backend constructed without any identification key (we should point this out explicitely in the docs though, maybe adding a reference to the AWS page showing how to set up your account), but I wonder if it would be possible to switch between different accounts by overwriting the environment variables or directly passing the key/token to the `device`.

Indeed, I'll go through the documentation to check again. Actually the account part is fairly complicated to me. Perhaps let's wait for Cody's response to clarify. I'll ping him again soon in case he misses this.

Reply:
Sorry for the late response! I actually would prefer the interface to use an ARN string instead of a device instance; in this case, it would be even better to have separate clients for `AwsDevice` (devices managed by the Amazon Braket service) and local devices.

As for setting up AWS credentials, yes, that's [correct](https://github.com/amazon-braket/amazon-braket-sdk-python?tab=readme-ov-file#iam-user-or-role-with-required-permissions) (except you only need to install `amazon-braket-sdk`).

Reply:
Hi @speller26 ! Just to clarify, is your suggestion to pass in 
``` python
device = "arn:aws:braket:::device/quantum-simulator/amazon/sv1"
```
instead of this
``` python
device = AwsDevice("arn:aws:braket:::device/quantum-simulator/amazon/sv1")
```
?
If so, then I think we're on the same page. I have made the changes. 

As for handling `AwsDevice` and local devices, what might be the benefit of having separate clients? I understand that there is are two main local devices, `LocalSimulator("default")` and`LocalSimulator("braket_dm")` but it feels easier to have them included in `BraketClientBackend` like what we did. Maybe I'm missing something.